bot/exts/utils/reminders.py

```diff
@@ -30,6 +32,133 @@
+        "30 minutes": 30 * 60,
+        "1 hour": 60 * 60,
+        "3 hours": 3 * 60 * 60,
+        "6 hours": 6 * 60 * 60,
+        "1 day": 24 * 60 * 60,
+        "1 week": 7 * 24 * 60 * 60,
+        "1 month": None,
+        "3 months": None,
+        "6 months": None,
+        "1 year": None
+    }
+
+    def __init__(self, bot: Bot, reminder_id: int, reminder_author_id: int):
+        super().__init__(timeout=30)
+        self.new_expiry: t.Optional[datetime] = None
+        self.reminder_was_snoozed = False
+
+        self.bot = bot
+        self.reminder_id = reminder_id
+        self.interaction_owner_id = reminder_author_id
+
+        self.dropdown: discord.ui.Select = self.children[0]
+        for duration in self.SNOOZE_DURATIONS:
+            self.dropdown.add_option(label=duration)
+
+    async def interaction_check(self, interaction: discord.Interaction) -> bool:
+        """Ensure that the user clicking the button is the member who invoked the command."""
+        if interaction.user.id != self.interaction_owner_id:
+            await interaction.response.send_message(":x: This is not your reminder to react to!", ephemeral=True)
+            return False
+        return True
+
+    @discord.ui.select(placeholder="Select the snooze duration")
+    async def select_snooze_duration(self, _: discord.ui.Select, interaction: discord.Interaction) -> None:
+        """Drop down menu which contains a list of snooze durations one can choose."""
+        selected_duration = interaction.data["values"][0]
+        current_datetime = datetime.now()
+        if (num_seconds := self.SNOOZE_DURATIONS.get(selected_duration)) is None:
+            amount, unit = selected_duration.split()
+            if unit.startswith("month"):
+                num_seconds = (current_datetime + relativedelta(months=int(amount)) - current_datetime).total_seconds()
+
+            else:  # unit is years
+                num_seconds = (current_datetime + relativedelta(years=int(amount)) - current_datetime).total_seconds()
+
+        new_reminder_end_time = current_datetime + relativedelta(seconds=num_seconds)
+
+        await self.bot.api_client.patch(
+            f'bot/reminders/{self.reminder_id}',
+            json={'expiration': new_reminder_end_time.isoformat()}
+        )
+
+        self.new_expiry = new_reminder_end_time
+        new_end_timestamp = time.discord_timestamp(new_reminder_end_time, format=time.TimestampFormats.DAY_TIME)
+        original_embed = interaction.message.embeds[0]
+        new_embed = discord.Embed(
+            colour=discord.Colour.green(),
+            title=random.choice(POSITIVE_REPLIES),
+            description=f"Successfully snoozed reminder for {selected_duration}. New remind time: {new_end_timestamp}",
+        )
+        new_embed.set_footer(text=f"ID: {self.reminder_id}")
+        try:
+            await interaction.response.edit_message(embeds=[original_embed, new_embed], view=None)
+        except discord.NotFound:
```

Comment:
what exactly is the reason for this?

Reply:
I believe this may be a left-over try/except from when I was trying to debug a separate error.

The idea is that the message may be deleted before we edit it, but in reality I don't think this would happen.

Reply:
Unless anyone else has any issues with this, I think I'll just leave it. From what I can tell there's no harm in having this safety net.